File: easy/array/max_product/max_product.go
package main

import (
	"fmt"
	"math"
)

func maxProduct(nums []int) int {
	firstMax, secondMax := math.MinInt64, math.MinInt64
	for i := 0; i < len(nums); i++ {
		if nums[i] > firstMax {
			secondMax, firstMax = firstMax, nums[i]
		} else if nums[i] > secondMax {
			secondMax = nums[i]
		}
	}

	return (firstMax - 1) * (secondMax - 1)
}

func main() {
	fmt.Println("Solution 1 : ", maxProduct([]int{3, 4, 5, 2}))
	fmt.Println("Solution 2 : ", maxProduct([]int{1, 5, 4, 5}))
	fmt.Println("Solution 3 : ", maxProduct([]int{3, 7}))
	fmt.Println("Solution 4 : ", maxProduct([]int{10, 2, 5, 2}))
}

File: easy/array/max_product/max_product.py
class Solution(object):
    def maxProduct(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        first_max = max(nums)
        nums.remove(first_max)
        second_max = max(nums)
        return (first_max-1)*(second_max-1)


s = Solution()
print("Solution 1 : ", s.maxProduct([3, 4, 5, 2]))
print("Solution 2 : ", s.maxProduct([1, 5, 4, 5]))
print("Solution 3 : ", s.maxProduct([3, 7]))
print("Solution 4 : ", s.maxProduct([10, 2, 5, 2]))
